cogs/embeds.py

```diff
@@ -1,24 +1,142 @@
 import discord
 from discord.ext import commands
+import json
+import re
 
 
 class Embeds(commands.Cog):
+
     def __init__(self, bot: commands.Bot) -> None:
         self.bot = bot
-    
+
     @commands.hybrid_command()
     async def embed(self, ctx: commands.Context):
         """
         Interactively construct a Discord embed.
         """
-        raise NotImplementedError('Command requires implementation and permission set-up.')
-    
+
+        def check(m: discord.Message):
+            return m.channel == ctx.channel and m.author == ctx.author
+
+        def is_skip(msg: discord.Message):
+            return msg.content.strip().lower() == "skip"
+
+        def is_done(msg: discord.Message):
+            return msg.content.strip().lower() == "done"
+
+
+        async def handle_color(embed: discord.Embed):
+            while True:
+                try:
+                    color_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+                    if is_skip(color_msg):
+                        break
+                    color_str = color_msg.content
+                    embed.color = discord.Color.from_str(color_str)
+                    break
+                except:
+                    await ctx.send("Invalid Color. Please provide the color in hex format (e.g. `#123456`). Use `skip` to skip this.")
+
+        async def handle_img(embed: discord.Embed):
+            img_url_msg: discord.Message = await self.bot.wait_for("message", check=check, timeout=timeout)
+            if is_skip(img_url_msg): return
+            img_url = img_url_msg.content.strip()
+            if img_url.startswith("<"): img_url = img_url[1:]
+            if img_url.endswith(">"): img_url = img_url[:-1]
+            try:
+                embed.set_image(url=img_url)
+            except:
```

Comment:
As before, always try to `except` a specific exception or error. In this case, [`Embed.set_image`](https://discordpy.readthedocs.io/en/stable/api.html?highlight=set_image#discord.Embed.set_image) does not raise any errors. However, it later raises an `HTTPException` in [`Context.send`](https://discordpy.readthedocs.io/en/stable/ext/commands/api.html#discord.ext.commands.Context.send). It would be good to move this try-except clause to the end of the function when `Context.send` is called.